backend/homework/8/src/main/java/com/example/demo2/controllers/VehicleController.java

```diff
@@ -0,0 +1,56 @@
+package com.example.demo2.controllers;
+
+import com.example.demo2.dto.VehicleDTO;
+import com.example.demo2.utility.Vehicle;
+import com.example.demo2.service.VehicleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/vehicles")
+/**
+ * This is the implementation of the APIs
+ */
+
+
+public class VehicleController {
+    @Autowired
+    private VehicleService vehicleService;
+
+    @GetMapping
+    public ResponseEntity<List<Vehicle>> getAllVehicles(){
+        List<Vehicle>vehicles = vehicleService.getVehicles();
+        return ResponseEntity.ok(vehicles);
+    }
+    @PostMapping
+    public ResponseEntity<Vehicle>createVehicle(@RequestBody VehicleDTO vehicleDTO){
+        Vehicle createdVehicle = vehicleService.createVehicle(vehicleDTO);
+        return ResponseEntity.ok(createdVehicle);
+    }
+    @PutMapping("/{name}")
```

Comment:
Should this have name as path variable or some ID ?
What if multiple names are same?